modin/pandas/indexing.py

```diff
@@ -821,22 +821,23 @@ def _setitem_with_new_columns(self, row_loc, col_loc, item):
                         "Must have equal len keys and value when setting with an iterable"
                     )
             else:
-                if item.shape != (len(self.qc.index, len(col_loc))):
+                if item.shape != (len(row_loc), len(col_loc)):
                     raise ValueError(
                         "Must have equal len keys and value when setting with an iterable"
                     )
-            exist_items = (
-                item[:, common_label_loc]
-                if len(item.shape) > 1
-                else item[common_label_loc]
-            )
+            if not isinstance(item, np.ndarray):
```

Comment:
At least in the case when the keys are strings, we will get an empty array here, which is not true.

Reply:
The previous comment is not entirely correct.

This part of the code was responsible for not taking those columns from the assigned value that were not in the query compiler, however this behavior is not true for `loc` operation, since missed columns will be added in the next few lines via `reindex` op.

Reply:
Considering also that in this code branch there can only be a numpy array, I completely deleted this code.